### ISSUE TITLE
Add note about sorting to documentation (and possibly a transform function)

### ISSUE THREAD

**Issue by theindigamer** [haskell/vector#232]
Continuing from #157 - sorting algorithms are available in `vector-algorithm`, but it isn't easy to find that information out, depending on how you search. For example, the Stackage Hoogle query for [sort :: Ord a => Vector a -> Vector a](https://www.stackage.org/lts-12.22/hoogle?q=sort+%3A%3A+Ord+a+%3D%3E+Vector+a+-%3E+Vector+a) doesn't come up with the result (because that package doesn't provide a wrapper for immutable vectors).

It would be nice if a note about this could be added to the top-level module `Data.Vector`, which serves as the "information hub" right now. Something along the lines of:

> Note: Sorting algorithms for mutable vectors are available in the [vector-algorithms](https://hackage.haskell.org/package/vector-algorithms) package. If you want to sort an immutable vector, you can use the [`freeze`](https://hackage.haskell.org/package/vector-0.12.0.2/docs/Data-Vector-Generic.html#v:freeze) and [`thaw`](https://hackage.haskell.org/package/vector-0.12.0.2/docs/Data-Vector-Generic.html#v:thaw) functions to wrap calls to sorting functions of your choice.

Personally, I think it would be nice to have a helper function alongside
```
Data.Vector.Generic.transform 
  :: (PrimMonad m, Vector v a) 
  => (Mutable v (PrimState m) a -> m (Mutable v (PrimState m) a)) 
  -> v a
  -> m (v a)
transform f = unsafeFreeze <=< f <=< thaw
```
as an counterpart to [Data.Vector.Generic.Mutable.transform](https://www.stackage.org/haddock/lts-12.9/vector-0.12.0.1/Data-Vector-Generic-Mutable.html#v:transform), but of course, it isn't strictly necessary, just a nice to have thing.

**Comment by cartazio** [comment]
this idea is growing on me, but i still need to think about it :)

**Comment by Shimuuar** [comment]
Yes sort of array _is_ well hidden and isn't very useable. Just a datapoint. `statistics` has helper for sorting: `sort :: Vector Double -> Vector Double`. In many cases convenience outweight cost of extra allocation or you have to allocate new vector anyway.

**Comment by Gabriel439** [comment]
I'd go a step further and suggest adding a default opinionated and pure `sort` directly to the `vector` package, with the documentation for this pure `sort` directing people to the `vector-algorithms` package if they are exceedingly performance conscious

**Comment by cartazio** [comment]
i'll have a look see, i do agree this might be nice, but want it to be well done :)
